doc/extdev/howto.rst

```diff
@@ -0,0 +1,161 @@
+Extension How-tos
+=================
+
+This section contains several how-tos associated with developing extensions in
+Sphinx.
+
+Discover builders by entry point
+--------------------------------
+
+.. versionadded:: 1.6
+
+:term:`Builder` extensions can be discovered by means of `entry points`_ so
+that they do not have to be listed in the :confval:`extensions` configuration
+value.
+
+Builder extensions should define an entry point in the ``sphinx.builders``
+group. The name of the entry point needs to match your builder's
+:attr:`~.Builder.name` attribute, which is the name passed to the
+:option:`sphinx-build -b` option. The entry point value should equal the
+dotted name of the extension module. Here is an example of how an entry point
+for 'mybuilder' can be defined in the extension's ``setup.py``::
+
+    setup(
+        # ...
+        entry_points={
+            'sphinx.builders': [
+                'mybuilder = my.extension.module',
+            ],
+        }
+    )
+
+Note that it is still necessary to register the builder using
+:meth:`~.Sphinx.add_builder` in the extension's :func:`setup` function.
+
+.. _entry points: https://setuptools.readthedocs.io/en/latest/setuptools.html#dynamic-discovery-of-services-and-plugins
+
+
+Defining custom template functions
+----------------------------------
+
+Sometimes it is useful to define your own function in Python that you wish to
+then use in a template. For example, if you'd like to insert a template value
+with logic that depends on the user's configuration in the site, or if you'd
```

Comment:
In Sphinx, we call whole of documents as "project" or "document". "site" is also good. But it is not appropriate on building PDF. I prefer to call it "project".